examples/metalk8s-solution-example/operator/pkg/config/config.go

```diff
@@ -0,0 +1,153 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+
+	"gopkg.in/go-playground/validator.v9"
+	"gopkg.in/yaml.v2"
+)
+
+type Repository struct {
+	Endpoint string `yaml:"endpoint" validate:"required"`
+	Images   []string `yaml:"images" validate:"required,ne=0"`
+}
+
+type OperatorConfig struct {
+	Kind         string `yaml:"kind" validate:"required,eq=OperatorConfig"`
+	APIVersion   string `yaml:"apiVersion" validate:"required"`
+	Repositories map[string][]Repository `yaml:"repositories" validate:"required,ne=0"`
+}
+
+var operatorConfig OperatorConfig
```

Comment:
Kept the approach with a global (but made it private to limit the risk) because we couldn't find a way to thread something from the main to the reconciliation loop (the loop has access to a `reconcile.Reconciler` and the only thing we get when creating it is the `manager.Manager`)

If you have a better suggestion, we're open for it.

Reply:
After discussing with @benoit-a I've implemented [this](https://github.com/scality/metalk8s/compare/bugfix/fix-example-solution...user/slaperche/fix-example-solution?expand=1) which allows us to get rid of the global config and thread the config variable from the main to the reconciler.

But this require some modification in the code generated by the SDK, thus I don't know if that's OK for you @gdemonet / @NicolasT or if you prefer the "private global" approach?
Or if you have a 3rd way :slightly_smiling_face: ?

Reply:
I think your Solution is better and we can change the code generated, it's not a problem, but I'm not a Go expert, so waiting for feedback from them.

Reply:
Yeah it's not a problem, except that it may make the upgrade of the SDK a bit more tedious (but  it's already a very manual process).

Reply:
I've cherry-picked your commit.